### PR TITLE
chore(influxdb_metrics sink): Upgrade `hyper`

### PR DESCRIPTION
**Pull request by ktff**
Ref. #2117

### CODE REVIEW

File: src/sinks/influxdb/metrics.rs [hunk 1]
```diff
@@ -128,10 +131,12 @@ impl InfluxDBSvc {
 impl Service<Vec<Metric>> for InfluxDBSvc {
     type Response = HttpResponse;
     type Error = HttpError;
-    type Future = Box<dyn Future<Item = Self::Response, Error = Self::Error> + Send + 'static>;
+    type Future = Pin<
+        Box<dyn std::future::Future<Output = Result<Self::Response, Self::Error>> + Send + 'static>,
+    >;
```
Comment (fanatid) [inline src/sinks/influxdb/metrics.rs:136]: Can be shortened with `BoxFuture` to:
`type Future = futures::future::BoxFuture<'static, Result<Self::Response, Self::Error>>`

File: src/sinks/influxdb/metrics.rs [hunk 2]
```diff
@@ -95,13 +98,13 @@ impl InfluxDBSvc {
...
-            let mut builder = hyper::Request::builder();
-            builder.method(Method::POST);
-            builder.uri(uri.clone());
-
-            builder.header("Content-Type", "text/plain");
-            builder.header("Authorization", format!("Token {}", token));
-            builder.body(body).unwrap()
+            hyper13::Request::builder()
+                .method(Method::POST)
+                .uri(uri.clone())
```
Comment (fanatid) [inline src/sinks/influxdb/metrics.rs:103]: - This is ok as in other PR, but can be shortened to `hyper13::Request::post(uri)` (in `mod.rs` this form used)
- No need clone `uri`

Comment (ktff) [reply src/sinks/influxdb/metrics.rs:103]: True. Those are relics from older code.

Comment (fanatid) [reply src/sinks/influxdb/metrics.rs:103]: Ah, I see now. Skipped that closure `build_request` used in service.

**Comment by fanatid** [comment]
Looks good, can we merge it?

**Comment by binarylogic** [comment]
@kirillt once you get an approval form the code owner (if there is one), or a team member, you can merge. Use your best judgment. If the change is complex it might warrant multiple reviewers.

**Comment by binarylogic** [comment]
Ignore me, I didn't realize @ktff authored this PR. Yes, we can merge if everyone approves and there are no pending changes.

**Pull request merged**